doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -9,80 +9,297 @@ The git repository for the project is hosted on Github at
+
+Contributors should be reasonably confident in the testing done on their code
+before asking for a review.  Should additional testing resources be needed, you
+can request help from a member of the core Chapel team when creating your pull
+request.
+
+.. _Testing guidance:
+
+Testing your patch:
+
+* Changes to the Chapel implementation should not cause regressions. Developers
+  are responsible for doing a degree of testing that's appropriate for their
+  change (described in the following bullets) and then can rely on nightly
+  regression testing to worry about the full cross-product of configurations.
+
+  * At a minimum, patches should pass correctness testing for the full test/
+    directory hierarchy for:
+
+    * ``CHPL_*_PLATFORM=linux64``
+    * ``CHPL_*_COMPILER=gnu``
+    * ``CHPL_COMM=none``
+    * ``CHPL_TASKS=<default>``
+
+  * Most developers will start by focusing on a subdirectory of tests that
+    exercise the features they changed, or test/release/ as a suite of tests
+    that exercises a rich and important slice of the language.
+
+  * Changes that are likely to affect multi-locale executions should also be
+    tested against tests that exercise multi-locale capabilities with
+    ``CHPL_COMM=gasnet``.  A common subset is: ``test/release/``,
+    ``test/multilocale/``, and ``test/distributions/``.
+
+  * Changes that are likely to cause portability issues should be tested against
+    different platforms and compilers to avoid fallout in the nightly testing to
+    the extent possible.
+
+* Note that the quickest way to do testing is to use the parallel testing system
+  across a large number of workstations.  If you have limited testing resources
+  available to you, you can request that a member of the core Chapel team help.
+
+.. _Test System: https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/TestSystem.rst
+
+
+.. _Request feedback on your changes:
+
+Request feedback on your changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. _Push your work:
+
+Push your work to your feature branch
++++++++++++++++++++++++++++++++++++++
+
+Push your changes to your feature branch on GitHub
+
+.. code-block:: bash
+
+    git push origin <branch_name>
+
+    # or if you don't like typing your complicated branch name,
+    # you can use this command to push the current branch:
+    git push origin HEAD
+
+    # if you forgot your branch name, you can get it by running
+    git branch
+
+    # it is the starred one...
+
+Note that if you have already created a pull request from a
+feature branch, pushing your work to that feature branch will
+update the pull request.
+
+This will enable others to see your work.
+
+.. _Ask for feedback early:
+
+Ask for feedback on your branch early (optional)
+++++++++++++++++++++++++++++++++++++++++++++++++
+
+Not ready to merge yours changes, but still want to see if your work is going
+in the right direction?  Feel free to ask for early feedback!
```

Comment:
Personally, I'm not as concerned with the how, but it seems reasonable to bring up the various strategies.

It might discourage contributions/interactions to require a WIP PR instead of allowing that option and pointing someone to just the branch - some people might be okay showing one person their unfinished branch but not having it visible on main repo until it is in a better state, for instance.